src/widgets/radio.jsx

```diff
@@ -574,17 +643,32 @@ var RadioEditor = React.createClass({
 
     onDelete: function(choiceIndex, e) {
         e.preventDefault();
+
         var choices = this.props.choices.slice();
+        var deleted = choices[choiceIndex];
+
         choices.splice(choiceIndex, 1);
-        this.props.onChange({choices: choices});
+
+        this.props.onChange({
+            choices: choices,
+            hasNoneOfTheAbove: !( deleted.isNoneOfTheAbove || !this.props.hasNoneOfTheAbove )
+        });
     },
 
-    addChoice: function(e) {
+    addChoice: function(noneOfTheAbove, e) {
         e.preventDefault();
 
         var choices = this.props.choices;
-        this.props.onChange({choices: choices.concat([{}])}, () => {
-            this.refs["editor" + choices.length].focus();
+        var newChoice = { isNoneOfTheAbove: !!noneOfTheAbove }
+        var insertIndex = choices.length - ( this.props.hasNoneOfTheAbove | 0 );
```

Comment:
Oops, I obviously didn't read this closely enough! I thought we were defaulting hasNoneOfTheAbove in case it was `undefined`.

For converting to an integer, I think `this.props.hasNoneOfTheAbove ? 1 : 0` is clearest, and speed isn't very important here. (Although currently for me, Chrom 44 is showing that as being the fastest on that jsperf test, which is surprising! I always though `|` was the fastest too.)